api/src/main/java/io/opentelemetry/trace/SpanId.java

```diff
@@ -48,7 +48,13 @@
   // The internal representation of the SpanId.
   private final long id;
 
-  private SpanId(long id) {
+  /**
+   * Constructs a {@code SpanId} whose representation is specified by a long value.
+   *
+   * @param id the long represenation of the {@code TraceId}.
+   * @since 0.1.0
+   */
+  public SpanId(long id) {
```

Comment:
Should we do any validation here? (e.g check for non-positive)

Reply:
This is mainly the reason why I thought this should not be in the public API. The SpanId is a byte array with a length 8. It just happens to us that we use long to represent this to avoid an extra allocation.

Reply:
What I tried to say is that people will get confused about this as @songy23 did.

Reply:
We can always add a clarification here, I think. **However**, if you think it's truly safer to not expose this, lets close this PR for now, and reconsider this later on if/as needed.

Reply:
I think it needs clear explanation. I do agree that we need this because we want to avoid extra allocations when generating new ids.